From: gangmo_seong <gangmo_seong@tmax.co.kr>
Date: Fri, 28 May 2021 11:23:33 +0900
Subject: [bugfix] setallocationmap error fix

- error fixed which setallocationmap() maps nothing to NOTI_CENTER

[IMS] N/A
---
 extensions/dash-to-panel/panel.js | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/extensions/dash-to-panel/panel.js b/extensions/dash-to-panel/panel.js
index 6b55d47..148b5d7 100644
--- a/extensions/dash-to-panel/panel.js
+++ b/extensions/dash-to-panel/panel.js
@@ -620,7 +620,6 @@ var dtpPanel = Utils.defineClass({ // eslint-disable-line no-unused-vars
 
         panelPositions.forEach(pos => {
             let allocationMap = this.allocationMap[pos.element];
-
             if (allocationMap.actor) {
                 allocationMap.actor.visible = pos.visible;
 
@@ -904,21 +903,20 @@ var dtpPanel = Utils.defineClass({ // eslint-disable-line no-unused-vars
         setMap(Pos.DATE_MENU, this.statusArea.dateMenu.container);
         setMap(Pos.SYSTEM_MENU, this.statusArea.aggregateMenu.container);
         setMap(Pos.RIGHT_BOX, this._rightBox, 1);
-        // Check whether notificationcenter extension is enabled
+        // Check whether notificationcenter loaded
         if (!this.isStandalone && this.statusArea.NotificationCenter)
             setMap(Pos.NOTI_CENTER, this.statusArea.NotificationCenter.container);
-        // NotificationCenter is not displayed on additional monitor
-        if (this.isStandalone && this.statusArea.NotificationCenter)
+        // If notificationcenter is not loaded, set temporary element
+        else
             setMap(Pos.NOTI_CENTER, this._dummynoti);
 
-        // setmap when notificationcenter loaded after dtpPanel
+        // When notification center loaded, replace dummy to notification center
         this.setmap_notificationcenter = Main.extensionManager.connect('extension-state-changed', (data, extension) => {
             if (extension.uuid !== 'notification-center@tmax-shell-extensions')
                 return;
             if (!this.isStandalone)
+            // Load notification center only in main monitor
                 setMap(Pos.NOTI_CENTER, this.statusArea.NotificationCenter.container);
-            else
-                setMap(Pos.NOTI_CENTER, this._dummynoti);
             // apply updated allocationMap to panel
             this.updateElementPositions();
         });
